src/main/kotlin/step2/SimpleExpressionParser.kt

```diff
@@ -0,0 +1,60 @@
+package step2
+
+class SimpleExpressionParser : ExpressionParser {
+
+    private val operands: MutableList<Int> = mutableListOf()
+    private val operators: MutableList<Operator> = mutableListOf()
+
+    override fun parse(expression: String) {
+        resetParser()
+
+        validateExpression(expression)
+        val tokenizedExpression = tokenizeExpression(expression)
+
+        for (i in 0..tokenizedExpression.lastIndex step 2) {
+            val token = tokenizedExpression[i]
+            val operand = token.toIntOrNull()
+            requireNotNull(operand)
+            require(!(operators.lastOrNull() == Operator.DIVIDE && operand == 0)) { "divide by zero" }
+            operands.add(operand)
+        }
+
+        for (i in 1..tokenizedExpression.lastIndex step 2) {
```

Comment:
[코틀린 컨벤션](https://kotlinlang.org/docs/coding-conventions.html#loops-on-ranges)에 따르면 for 문 사용 시 Range를 그대로 사용하는 것 보단 `until`을 사용해 표현해주는 걸 권장하고 있어요.
```suggestion
        for( i in 0 until tokenizedExpression.size step 2) {
```
또는 범위가 지정된 Range라면 다음과 같이 indice 프로퍼티를 활용해주셔도 좋을 것 같습니다!
```kotlin
 for( i in tokenizedExpression.indices step 2) {
```